src/Connections.Abstractions/DefaultConnectionContext.cs

```diff
@@ -65,7 +65,9 @@ public DefaultConnectionContext(string id, IDuplexPipe transport, IDuplexPipe ap
 
         public CancellationToken ConnectionClosed { get; set; }
 
-        public virtual void Abort()
+        public void Abort() => Abort(abortReason: null);
```

Comment:
Ugh, if we're going to make a breaking change, I'd like this to be moved to ConnectionContext.

Reply:
Does SignalR ever directly implement ConnectionContext in its tests or anything like that? I want to know if, unlike the IConnectionLifetimeFeature breaking change, this could actually break anyone downstream. @anurse

Reply:
It would be added as virtual so non-breaking.

Reply:
Yuck. https://github.com/aspnet/KestrelHttpServer/pull/2612/files#r191886723

To continue with this "if we're going to make a breaking change" mindset, do you think we should remove both Abort methods from IConnectionLifetimeFeature and change IConnectionLifetimeFeature.ConnectionClosed to a Task since it always completes eventually so we don't have to worry about the transports waiting for the middleware to complete prior to disposing the ConnectionClosed CancellationToken?

Reply:
If you're committed to making Abort virtual, should ConnectionContext.Abort throw a NotImplementedException? I don't see how that's better than a MethodMissingException, but at least you wouldn't end up silently leaking sockets.

Reply:
We do have a `TestConnectionContext` that inherits from it but it should be fine, it should just inherit it.

Reply:
FWIW, you could probably just use Microsoft.AspNetCore.Connections.DefaultConnectionContext instead of using your own TestConnectionContext.

Reply:
Yeah, we do that in other places, not entirely sure why we do it there, I'll take a quick scan.